lms/djangoapps/api_manager/courses/tests.py

```diff
@@ -2148,6 +2148,80 @@ def test_courses_data_metrics(self):
         response = self.do_get(course_metrics_uri)
         self.assertEqual(response.status_code, 404)
 
+    def test_course_data_metrics_user_group_filter_for_empty_group(self):
+        group = GroupFactory.create()
+
+        # get course metrics for users in group
+        course_metrics_uri = '{}/{}/metrics/?groups={}'.format(
+            self.base_courses_uri,
+            self.test_course_id,
+            group.id
+        )
+        response = self.do_get(course_metrics_uri)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['users_enrolled'], 0)
+        self.assertGreaterEqual(response.data['users_started'], 0)
+
+    def test_course_data_metrics_user_group_filter_for_group_having_members(self):
+        group = [GroupFactory.create()]
+        users = UserFactory.create_batch(3, groups=group)
+
+        # enroll all users in course
+        for user in users:
+            CourseEnrollmentFactory.create(user=user, course_id=self.course.id)
+
+        # create course completions
+        for user in users:
+            completions_uri = '{}/{}/completions/'.format(self.base_courses_uri, self.test_course_id)
+            completions_data = {
+                'content_id': unicode(self.course_content.scope_ids.usage_id),
+                'user_id': user.id,
+                'stage': 'First'
+            }
+            response = self.do_post(completions_uri, completions_data)
+            self.assertEqual(response.status_code, 201)
+
+        course_metrics_uri = '{}/{}/metrics/?groups={}'.format(
+            self.base_courses_uri,
+            self.test_course_id,
+            group[0].id
+        )
+        response = self.do_get(course_metrics_uri)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['users_enrolled'], 3)
+        self.assertGreaterEqual(response.data['users_started'], 3)
+
+    def test_course_data_metrics_user_group_filter_for_multiple_groups_having_members(self):
+        groups = GroupFactory.create_batch(2)
+        users = UserFactory.create_batch(4, groups=(groups[0],))
+        users.append(UserFactory.create(groups=groups))
```

Comment:
adding this separate user to both groups while rest of the users are in group one only.